catalog/core/catalog-core-api/src/main/java/ddf/catalog/plugin/PolicyPlugin.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.plugin;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.Query;
+
+/**
+ * A PolicyPlugin is used to build policy information regarding the Catalog action a user is attempting.
+ */
+public interface PolicyPlugin {
+
+    String OPERATION_SECURITY = "operation.security";
+
+    /**
+     * Processes a {@link Metacard}, prior to ddf.catalog.source.CatalogProvider#create(CreateRequest), to return policy information
+     *
+     * @param input the {@link Metacard} to process
```

Comment:
I'm going to skip this one for now. If these plugin classes were off in their own API bundle, I think I'd have less of a problem doing it. If I start marking things in this API as not null, someone not familiar with the code might thing that means that things that aren't marked with `@NotNull` *can* be null, and that just isn't true. So, in my opinion, we either need to go through all the interfaces and mark them, or leave them as is. The other issue I have is that javax.validation is actually part of Java EE, which we aren't using, and that feels like we'd be misusing that annotation (but that is just my opinion :D )